controllers/api.controllers.js

```diff
@@ -1,13 +1,39 @@
-const { selectTopics } = require('../models/api.models');
+
+const { selectTopics, selectArticle } = require("../models/api.models");
+const endpoints = require("../endpoints.json");
 
 const getTopics = (req, res, next) => {
-    selectTopics()
-        .then(topics => {
-            res.status(200).send(topics);
-        })
-        .catch(err => {
-            next(err); 
-        });
+  selectTopics()
+    .then((topics) => {
+      res.status(200).send(topics);
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+
+module.exports = {getTopics}
+
+const getApi = (req, res, next) => {
+  res.status(200).send({ endpoints });
+};
+
+const getArticle = (req, res, next) => {
+  const { article_id } = req.params;
+  if (!Number.isInteger(parseInt(article_id))) {
```

Comment:
We do not need to check if is an integer here because psql will take care of these type of issues, refer to the notes  under error handling section or the can-you-handle-this sprint to create an error handler middleware for psql errors 🙂